### DISCUSSION TITLE
Will be trying to run portal local in a windows 11 machine.

### DISCUSSION THREAD

**Discussion by rajeevsathish** [Sunbird-Ed/Community#433] [General Discussion]
Will be trying to run portal local installation in a windows 11 Machine using the document
from 
https://ed.sunbird.org/use-1/install-locally/sunbirded-portal

CC: @vinukumar-vs @ravishtrigyn @Krishnaj20

**Comment by rajeevsathish** [comment]
For using windows with node we need some prerequisites to get it setup in the windows and the steps are as follows.

1. Install Python latest (3.11.2) from https://www.python.org/downloads/


![python](https://user-images.githubusercontent.com/5653717/225547160-ba6127e7-a80c-43c0-ac15-e49c89d082fd.png)

2. Update the python path in the windows environment variable path

![envvar](https://user-images.githubusercontent.com/5653717/225547477-da28b82d-aba7-4768-9cf6-1c592ec2db05.png)

3. Install NVM to support multiple version of Node from
path https://github.com/coreybutler/nvm-windows#installation--upgrades

![nvm](https://user-images.githubusercontent.com/5653717/225547707-cedb1e5b-c6f8-49c9-ab06-8f92dbecef45.png)

4. Install node 14 using nvm command in cmd by opening with admin mode

      `nvm install v14` 

post installation run nvm use command

       `nvm use v14`

5. check npm config using the command

      `npm config list`

   this should list the config like 
   
![config](https://user-images.githubusercontent.com/5653717/225548423-75e6899d-89f7-422d-a77a-44ec1e41091b.png)

if not you need to add the missing config
            5 a. installing node-gyp by using the command
                          `npm install -global node-gyp`
                   set node-gyp path in the npm config using command
				`npm config set node-gyp <executable path of the node-gyp javascript>`
            5 b. install microsoft visual studio 2022 from https://visualstudio.microsoft.com/downloads/
            
![config](https://user-images.githubusercontent.com/5653717/225551432-edca4ace-61e6-4992-8835-e87a3c48b07c.png)

 for more details please refer this link
 https://stackoverflow.com/questions/70082657/npm-err-gyp-err-when-installing-node-sass-in-windows-11-with-visual-studio-202
                    post installation set the msvs version in the npm config using the command
				`npm config set msvs_version 2022`
             5 c. update the npm config using the command
				`npm config set python <executable path of the python>
`post this install the sunbird using the document mentioned here
    https://ed.sunbird.org/use-1/install-locally/sunbirded-portal
cc: @vinukumar-vs @Krishnaj20 @chitranshu-keshav @ravishtrigyn

**Comment by ravishtrigyn** [comment]
@rajeevsathish Did it build successfully with Node 14? When I downgraded to Node 14 from 16, my client did not even build. With 16 it was atleast building.

**Reply by rajeevsathish** [reply]
please refer to the screen shots given 
the node Version
![Screenshot (7)](https://user-images.githubusercontent.com/5653717/227843816-e4621af3-2d4f-41e3-8985-f98bd008085b.png)
the Angular version
![Screenshot (8)](https://user-images.githubusercontent.com/5653717/227843922-3cb937ff-5e88-4000-ac71-2b66ffe319b5.png)
and the branch
![Screenshot (9)](https://user-images.githubusercontent.com/5653717/227844004-6d4f4349-aff7-459a-82e8-df2807774c7b.png)
